docs/building.md

```diff
@@ -91,7 +91,9 @@ $ cd cpp-sc2
 $ cmake -B build
 
 # Build the project.
-$ cmake --build build --parallel
+# Note: number after '-j' is threads used.
+# Max is 'n-1' threads your cpu supports.
+$ cmake --build build --parallel -j 4
```

Comment:
Do we have this issue only in Linux? At least macOs has the same problem.

Also, we can try to use all available core - 1 e.g. by doing:
https://unix.stackexchange.com/questions/208568/how-to-determine-the-maximum-number-to-pass-to-make-j-option

Reply:
The following works on mac:

```bash
$ nproc
12

$ nproc --ignore=1
11
```

Reply:
> From what I read, it's not just about jobs, but the memory it uses. So when google uses 9 jobs, it is over 13.5 GBs on my machine?)

Ah, that could be. Nice catch!

Ok, lets do the following:
1. For macOs and Linux add `$(nproc)`.
2. For windows put `4`, but add comment that one can increase this value according to his cpu count.

Reply:
Addressed in https://github.com/cpp-sc2/cpp-sc2/commit/e4e7bafca81d5266da811370bf44503729a0fd41 with suggested corrections.

Thank you for the patch!